core/src/main/java/com/oath/oak/Handle.java

```diff
@@ -58,11 +61,11 @@ boolean put(V newVal, OakSerializer<V> serializer, MemoryManager memoryManager)
             return false;
         }
         int capacity = serializer.calculateSize(newVal);
-        if (this.value.remaining() < capacity) { // can not reuse the existing space
+        if (this.value.capacity() < capacity) { // can not reuse the existing space
             memoryManager.release(this.value);
             this.value = memoryManager.allocate(capacity);
         }
-        serializer.serialize(newVal, this.value.slice());
+        serializer.serialize(newVal, this.value.nioBuffer());
```

Comment:
I wonder about the usage of nioBuffer()... I wonder if this is what we want, and do we follow the original intent if switch to such usage...

This is from its description:

public abstract java.nio.ByteBuffer nioBuffer()
Exposes this buffer's readable bytes as an NIO ByteBuffer. The returned buffer either share or contains the copied content of this buffer, while changing the position and limit of the returned NIO buffer does not affect the indexes and marks of this buffer. This method is identical to buf.nioBuffer(buf.readerIndex(), buf.readableBytes()). This method does not modify readerIndex or writerIndex of this buffer. Please note that the returned NIO buffer will not see the changes of this buffer if this buffer is a dynamic buffer and it adjusted its capacity.
Throws:
java.lang.UnsupportedOperationException - if this buffer cannot create a ByteBuffer that shares the content with itself

Reply:
Take a look at the description here: https://netty.io/4.0/api/io/netty/buffer/ByteBuf.html
You are write that this returns a sliced bytebuffer from readerIndex to writerIndex, thats why in line 37 i do value.writerIndex(value.capacity()); to move the writerIndex to the end.

Reply:
Why do I think that usage of nioBuffer() might not be correct/efficient is from the following parts of the description:

Exposes this buffer's **readable bytes** as an NIO ByteBuffer. The returned buffer either share **or contains the copied content** of this buffer, **while changing the position and limit of the returned NIO buffer does not affect the indexes and marks of this buffer**.

1. "readable bytes" what if nothing was yet written into this buffer and readable bytes are actually nothing?
2. "contains the copied content" if it goes to copy the content (even in some specific cases) it is not good solution for us
3. "changing the position and limit of the returned NIO buffer does not affect the indexes and marks of this buffer" it looks like the returned buffer was intend to be read-only and if we let user write into it (and we do), the write index will not be updated and user might not see the changes on next read...